bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/BahmniPatientService.java

```diff
@@ -18,4 +18,13 @@
     public List<Patient> get(String partialIdentifier, boolean shouldMatchExactPatientId);
 
     public List<RelationshipType> getByAIsToB(String aIsToB);
+
+    /**
+     * Intersection of lucene and hibernate search in one query
+     * First take advantage of lucene and fall back to hibernate
+     * @param searchParameters
+     * @return List of PatientResponse
+     */
+    List<PatientResponse> luceneHibernateSearch(PatientSearchParameters searchParameters);
```

Comment:
Why does this method name need to say this. It's possible that the under-the-hood implementation does in fact use lucene and then hibernate, but is that actually part of the interface contract? Isn't this just "searchBy" or something like that?